error.go

```diff
@@ -0,0 +1,35 @@
+package main
```

Comment:
Minor nitpick: Go already has a built-in type called `error`, so calling the file `error.go` is a little confusing. Since the type you have inside is `fetchError`, perhaps name the file the same?